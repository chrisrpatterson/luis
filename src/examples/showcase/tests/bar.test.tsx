import * as React from 'react';
import * as renderer from 'react-test-renderer';

import * as PropTypes from 'prop-types';
import { load } from '../../../proxies/loader';

describe('Component', () => {
  describe('With Context', function() {
    const Component: React.FC = ({}, { theme }) => {
      return (
        <div style={{ backgroundColor: theme.backgroundColor, color: theme.color }}>
          Bar Component 333
        </div>
      );
    };

    // test
    Component.contextTypes = {
      theme: PropTypes.object.isRequired
    };

    it('fails', function() {
      throw new Error('Failed miserably');
    });

    const config = {
      component: Component,
      context: {
        theme: {
          backgroundColor: '#f1f1f1',
          color: '#222'
        }
      }
    };

    it('renders', function() {
      expect(renderer.create(load(config))).toMatchSnapshot();
    });

    return config;
  });
});
