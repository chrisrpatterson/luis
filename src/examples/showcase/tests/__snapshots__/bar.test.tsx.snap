// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component With Context renders 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#f1f1f1",
      "color": "#222",
    }
  }
>
  Bar Component
</div>
`;
